src/helpers/format.ts

```diff
@@ -1,10 +1,58 @@
 import type { ReactTestRendererJSON } from 'react-test-renderer';
-import prettyFormat, { plugins } from 'pretty-format';
+import prettyFormat, { NewPlugin, plugins } from 'pretty-format';
 
-const format = (input: ReactTestRendererJSON | ReactTestRendererJSON[]) =>
+type FilterNodeByTypeFunction = (
+  propName: string,
+  propValue: unknown,
+  node: ReactTestRendererJSON
+) => boolean;
+
+export type FormatOptions = {
+  filterProps?: FilterNodeByTypeFunction;
+};
+
+const format = (
+  input: ReactTestRendererJSON | ReactTestRendererJSON[],
+  options: FormatOptions = {}
+) =>
   prettyFormat(input, {
-    plugins: [plugins.ReactTestComponent, plugins.ReactElement],
+    plugins: [getCustomPlugin(options.filterProps)],
```

Comment:
Seems like we might as well use the orginal ReactElement plugin and just tweak the ReactTestComponent
```suggestion
    plugins: [reactTestComponentPlugin(options.filterProps), plugins.ReactElement],
```